app/models/user.rb

```diff
@@ -0,0 +1,13 @@
+class EmailValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    pattern = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+    record.errors[attribute] << (options[:message] || 'is not an email') unless value =~ pattern
+  end
+end
+
+class User < ApplicationRecord
```

Comment:
- [ ] If you are grouping classes with related functionality into a single file then you should also consider making them part of the same module for a namespace. So it would be better to create a helper function to validate email instead of creating another class.